pysat/tests/instrument_tests.py

```diff
@@ -0,0 +1,239 @@
+    # directory
+    if len(inst.files.files.values) < 20:
+        for the_file in list(inst.files.files.values):
+            # Check if filename is appended with date for fake_daily data
+            # ie, does an underscore exist to the right of the file extension?
+            if the_file.rfind('_') > the_file.rfind('.'):
+                # If so, trim the appendix to get the original filename
+                the_file = the_file[:the_file.rfind('_')]
+            file_path = os.path.join(temp_dir, the_file)
+            if os.path.isfile(file_path):
+                os.unlink(file_path)
+    else:
+        warnings.warn(''.join(('Files > 20.  Not deleted.  Please check to ',
+                               'ensure temp directory is used')))
+
+
+def generate_instrument_list(instrument_names=[], package=None):
+    """Iterate through and create all of the test Instruments needed.
+       Only want to do this once.
+
+    """
+
+    if package == None:
+        package = 'pysat.instruments'
+
+    instrument_download = []
+    instrument_no_download = []
+
+    # create temporary directory
+    dir_name = tempfile.mkdtemp()
+    saved_path = pysat.data_dir
+    pysat.utils.set_data_dir(dir_name, store=False)
+
+    for name in instrument_names:
+        try:
+            module = import_module(''.join(('.', name)),
+                                   package=package)
+        except ImportError:
+            print(' '.join(["Couldn't import", name]))
+            pass
+        else:
+            # try to grab basic information about the module so we
+            # can iterate over all of the options
+            try:
+                info = module._test_dates
+            except AttributeError:
+                info = {}
+                info[''] = {'': pysat.datetime(2009, 1, 1)}
+                module._test_dates = info
+            for sat_id in info.keys():
+                for tag in info[sat_id].keys():
+                    try:
+                        inst = pysat.Instrument(inst_module=module,
+                                                tag=tag,
+                                                sat_id=sat_id,
+                                                temporary_file_list=True)
+                        inst._test_dates = module._test_dates
+                        travis_skip = ((os.environ.get('TRAVIS') == 'true')
+                                       and not inst._test_download_travis)
+                        if inst._test_download:
+                            if not travis_skip:
+                                instrument_download.append(inst)
+                        elif not inst._password_req:
+                            # we don't want to test download for this combo
+                            # But we do want to test the download warnings
+                            # for instruments without a password requirement
+                            instrument_no_download.append(inst)
+                    except:
```

Comment:
Comment here explaining catch-all except would be useful.

Reply:
I think the intent is that if an instrument fails to load, it is skipped so the rest will at least be tested.  However, we already check that the module can be imported before running this code at Line 78.  I think we can remove this try / except safely.